controllers/conditions.go

```diff
@@ -0,0 +1,133 @@
+package controllers
+
+import (
+	"context"
+	"time"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/image-based-install-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func findCondition(conditions []hivev1.ClusterInstallCondition, condType hivev1.ClusterInstallConditionType) *hivev1.ClusterInstallCondition {
+	for i := range conditions {
+		if conditions[i].Type == condType {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+func setClusterInstallCondition(conditions *[]hivev1.ClusterInstallCondition, newCondition hivev1.ClusterInstallCondition) {
+	if conditions == nil {
+		return
+	}
+	now := metav1.NewTime(time.Now())
+	existingCondition := findCondition(*conditions, newCondition.Type)
+	if existingCondition == nil {
+		if newCondition.LastTransitionTime.IsZero() {
```

Comment:
Curious, Who is setting the `LastProbeTime` and `LastTransitionTime` on the newCondition?
From the code it seems that they are never set except for here so why bother checking for IsZero() ?

Reply:
Ah this is just a copy of the core function that does the same thing, but with core conditions.

I can trim this down to only what we need.

Reply:
This is the function I was referencing FWIW https://github.com/kubernetes/apimachinery/blob/v0.28.3/pkg/api/meta/conditions.go#L30